support/modules/rootfs-image

```diff
@@ -81,6 +81,27 @@ set_upgrade_vars() {
     upgrade_available="${upgrade_available#upgrade_available=}"
 }
 
+check_environment_canary() {
+    mender_check_saveenv_canary="$(${PRINTENV} mender_check_saveenv_canary)"
+    if [ "$mender_check_saveenv_canary" = "mender_check_saveenv_canary=1" ]; then
+	# If the check canary exists (added during build), we need to check the real canary to make
+	# sure that the boot loader was successful in adding it during boot.
+	mender_saveenv_canary="$(${PRINTENV} mender_saveenv_canary)"
+	if [ "$mender_saveenv_canary" != "mender_saveenv_canary=1" ]; then
+	    cat 1>&2 <<'EOF'
+`mender_check_saveenv_canary` was set in the boot environment, but
```

Comment:
Careful on the backquotes

Reply:
I already got burned! 😄

That's why `'EOF'` is single-quoted, it disables all expansions.

Reply:
haha, I figured. 

I would still suggest single-quoting this instead though, and remove the verbatim expansion. but YMMV

Reply:
I don't think I want to introduce backslashes if we can avoid it, and we don't need expansions here, so I think I'll leave it as is.